.circleci/config.yml

```diff
@@ -160,10 +170,15 @@ jobs:
   build-osx:
     executor: macos
     steps:
-      - install_macos_dependencies
       - checkout
+      - run: |
+            if ! ./scripts/needs_ci_run; then
+                echo "Changes in last commit do not need CI. Skipping step"
+                circleci-agent step halt
+            fi
       - run: ./scripts/format.sh && ./scripts/error_on_dirty.sh
       - run: git submodule sync && git submodule update --init
+      - install_macos_dependencies
```

Comment:
so all the dependencies needed for `needs_ci_run`, `format.sh` and `error_on_dirty.sh` must already been installed by default? i guess its just python, since format.sh gets its stuff from mason? i thought error on dirty used `parallel` but i havent checked!

Reply:
`needs_ci_run` uses vanilla python modules, and it seems that the OSX executor comes installed with it. `error_on_dirty.sh` is not using `parallel`. `format.sh` does have `mason` as a dependency, but it also installs it if not already installed. This change is strictly not need (the install step does add 2-3 mins which is not too bad I guess?). Initially I was checking to see if the skip check would work if I moved `install_dependencies` to a later stage. It did, so I left the change in the PR.

Reply:
yep its all good, i was just refreshing my memory! wasnt asking for a change or anything like that. if we can save some time lets do it